### PR TITLE
Add the dev image used in mediawiki-containers

### PR DESCRIPTION
**Pull request by d00rman**

### CODE REVIEW

File: dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
+FROM debian:sid
```
Comment (addshore) [inline dev/Dockerfile]: This should probably be based on one of the mediawiki images, and simply add things such as xdebug etc.

File: dev/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+ENV MEDIAWIKI_VERSION wmf/1.30.0-wmf.4
```
Comment (addshore) [inline dev/Dockerfile]: There isn't really a reason to require a specific version in the image.
You can already take any of the current mediawiki images, mount your local copy of mediawiki (and extensions) over it and use it as a basic development environment.

**Comment by addshore** [comment]
The main issue I see with this image in its current state is how much it diverges from the other mediawiki images.

**Comment by d00rman** [comment]
> The main issue I see with this image in its current state is how much it diverges from the other mediawiki images.

This project initially started out as the foundation for the image used in mediawiki-containers. However, all the work in that direction was removed without our approval, so I'm merely committing back what was there in the first place.

**Comment by addshore** [comment]
Okay!

See https://phabricator.wikimedia.org/T92826#3518629.

Im not sure if keeping the images published to https://hub.docker.com/_/mediawiki/ and the base image for the mediawiki-containers work in the same repo makes sense, perhaps this commit should actually land in the mediawiki-containers repo?

I'm keen to have a development image in this repo and downloadable from https://hub.docker.com/_/mediawiki/

**Comment by d00rman** [comment]
@addshore Right now, the images coming from this Dockerfile are published as the [Wikimedia Mediawiki images](https://hub.docker.com/r/wikimedia/mediawiki/). While we call it a *dev* image, it is currently not. We are simply using it as-is in [mediawiki-containers](https://github.com/wikimedia/mediawiki-containers/blob/2e4f836853653ee2dd04381d4c3cb33115cdd6cf/mediawiki-dev.yaml#L99-L151). The plan is to work on it and turn it into a veritable dev image. Once that happens, I'd be +1 on publishing it under the `dev` tag in the main Docker Library, but not before that.

**Comment by addshore** [comment]
perhaps we should stick them in a wmf directory for now instead of dev?

**Comment by davidbarratt** [comment]
The problem with this is, is that we will **not** be able to push any new images to https://hub.docker.com/_/mediawiki/ because the whole repo will be reviewed by @docker-library.

We either need to have an official dev image (which I'm fine with, but we'd need to update it with every dev release) or we need to remove it.

Alternatively, we can move it to a different branch for now.

Leaving it on the `master` branch isn't going to work as soon as their is a MediaWiki release.

**Comment by addshore** [comment]
I propose we create a branch from the current HEAD of master, called dev.
Then revert this PR on master.

**Pull request merged**